aggregator/src/test/java/org/jenkinsci/plugins/workflow/steps/TimeoutStepTest.java

```diff
@@ -22,57 +24,101 @@
-                        ");",
-                        "echo 'NotHere'",
-                    "}",
-                    "echo 'NotHere'",
-                "}")));
-        WorkflowRun b = r.assertBuildStatus(/* TODO JENKINS-25894 should really be ABORTED */Result.FAILURE, p.scheduleBuild2(0).get());
+        story.addStep(new Statement() {
+            @Override
+            public void evaluate() throws Throwable {
+                WorkflowJob p = story.j.jenkins.createProject(WorkflowJob.class, "p");
+                p.setDefinition(new CpsFlowDefinition(""
+                        + "node {\n"
+                        + "  timeout(time:5, unit:'SECONDS') {\n"
+                        + "    parallel(\n"
+                        + "      a: { echo 'ShouldBeHere1'; sleep 10; echo 'NotHere' },\n"
+                        + "      b: { echo 'ShouldBeHere2'; sleep 10; echo 'NotHere' },\n"
+                        + "    );\n"
+                        + "    echo 'NotHere'\n"
+                        + "  }\n"
+                        + "  echo 'NotHere'\n"
+                        + "}\n"));
+                WorkflowRun b = story.j.assertBuildStatus(/* TODO JENKINS-25894 should really be ABORTED */Result.FAILURE, p.scheduleBuild2(0).get());
 
-        // make sure things that are supposed to run do, and things that are NOT supposed to run do not.
-        r.assertLogNotContains("NotHere", b);
-        r.assertLogContains("ShouldBeHere1",b);
-        r.assertLogContains("ShouldBeHere2", b);
+                // make sure things that are supposed to run do, and things that are NOT supposed to run do not.
+                story.j.assertLogNotContains("NotHere", b);
+                story.j.assertLogContains("ShouldBeHere1", b);
+                story.j.assertLogContains("ShouldBeHere2", b);
 
-        // we expect every sleep step to have failed
-        FlowGraphTable t = new FlowGraphTable(b.getExecution());
-        t.build();
-        for (Row r : t.getRows()) {
-            if (r.getNode() instanceof StepAtomNode) {
-                StepAtomNode a = (StepAtomNode) r.getNode();
-                if (a.getDescriptor().getClass()==SleepStep.DescriptorImpl.class) {
-                    assertTrue(a.getAction(ErrorAction.class)!=null);
+                // we expect every sleep step to have failed
+                FlowGraphTable t = new FlowGraphTable(b.getExecution());
+                t.build();
+                for (Row r : t.getRows()) {
+                    if (r.getNode() instanceof StepAtomNode) {
+                        StepAtomNode a = (StepAtomNode) r.getNode();
+                        if (a.getDescriptor().getClass() == SleepStep.DescriptorImpl.class) {
+                            assertTrue(a.getAction(ErrorAction.class) != null);
+                        }
+                    }
                 }
             }
-        }
+        });
     }
 
-    // TODO: timeout inside parallel
```

Comment:
@jglick Different coding styles. To compensate, I'll add a new test, ok? :smile: